mobile/src/main/java/ch/epfl/sdp/mobile/ui/game/ar/ChessScene.kt

```diff
@@ -31,89 +31,57 @@ import kotlinx.coroutines.flow.onEach
  * - The board as the root node
  * - A number of chess pieces depending on the game state
  *
- * @param context The context used to load the 3d models
  * @param scope A scope that is used to launch the model loading
- * @param startingBoard The board that contains the displayed game state
  */
 class ChessScene<Piece : ChessBoardState.Piece>(
     private val context: Context,
-    scope: CoroutineScope,
+    private val scope: CoroutineScope,
     startingBoard: Map<Position, Piece>,
 ) {
 
   /** The [ArModelNode] which acts as the root of the [ArModelNode] hierarchy. */
   val boardNode = ArModelNode(placementMode = PlacementMode.PLANE_HORIZONTAL)
 
+  private val currentPieces: MutableMap<Piece, ModelNode> = mutableMapOf()
+
   /** A conflated [Channel] which associates the position of the pieces to their values. */
   private val currentPositionChannel =
       Channel<Map<Position, Piece>>(capacity = CONFLATED).apply { trySend(startingBoard) }
 
+  // Board Bounding box
+  private var boundingBox: Box? = null
```

Comment:
This could probably be passed as an argument rather than stored as a field.

Reply:
Fix in https://github.com/epfl-SDP/android/pull/391/commits/a85ea39df65a1e748f84c3cb9cc34f0afeba14d4